knowageutils/src/main/java/it/eng/spagobi/tools/dataset/common/datareader/XmlDataReader.java

```diff
@@ -168,6 +170,23 @@ public IDataStore read(Object data) {
 		return dataStore;
 	}
 
+	private static DocumentBuilderFactory getFactory() {
+		DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+		try {
+			dbf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+			dbf.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+			dbf.setFeature("http://xml.org/sax/features/external-parameterentities", false);
+			dbf.setFeature("http://xml.org/sax/features/external-generalentities", false);
+		} catch (ParserConfigurationException e) {
+			logger.error("Error loading XML document: " + e.getMessage(), e);
+			throw new SecurityException("Error loading XML document: " + e.getMessage(), e);
```

Comment:
The error message is misleading: we are not loading XML document in that block

Reply:
Fixed